lib/usd/translators/shading/usdPhongReader.cpp

```diff
@@ -0,0 +1,128 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "usdLambertReader.h"
+
+#include <mayaUsd/fileio/shaderReader.h>
+#include <mayaUsd/fileio/shaderReaderRegistry.h>
+#include <mayaUsd/fileio/shading/shadingModeRegistry.h>
+#include <mayaUsd/fileio/translators/translatorUtil.h>
+#include <mayaUsd/fileio/utils/readUtil.h>
+#include <mayaUsd/utils/util.h>
+
+#include <pxr/base/tf/diagnostic.h>
+#include <pxr/base/tf/staticTokens.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/base/vt/value.h>
+#include <pxr/pxr.h>
+#include <pxr/usd/sdf/path.h>
+#include <pxr/usd/sdf/types.h>
+#include <pxr/usd/sdf/valueTypeName.h>
+#include <pxr/usd/usdShade/input.h>
+#include <pxr/usd/usdShade/output.h>
+#include <pxr/usd/usdShade/shader.h>
+#include <pxr/usd/usdShade/tokens.h>
+
+#include <maya/MFnDependencyNode.h>
+#include <maya/MObject.h>
+#include <maya/MPlug.h>
+#include <maya/MStatus.h>
+
+#include <basePxrUsdPreviewSurface/usdPreviewSurface.h>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+/// Shader writer for importing UsdPreviewSurface to Maya's lambert material nodes
+class PxrUsdTranslators_PhongReader : public PxrUsdTranslators_LambertReader {
+    typedef PxrUsdTranslators_LambertReader _BaseClass;
+
+public:
+    PxrUsdTranslators_PhongReader(const UsdMayaPrimReaderArgs&);
+
+    static ContextSupport CanImport(const UsdMayaJobImportArgs& importArgs);
+
+    /// Get the name of the Maya shading attribute that corresponds to the
+    /// USD attribute named \p usdAttrName.
+    virtual TfToken GetMayaNameForUsdAttrName(const TfToken& usdAttrName) const;
+
+protected:
+    /// What is the Maya node type name we want to convert to:
+    const TfToken& _GetMayaNodeTypeName() const override;
+
+    void _ConvertToMaya(const TfToken& mayaAttrName, VtValue& usdValue) const override;
+};
+
+PXRUSDMAYA_REGISTER_SHADER_READER(UsdPreviewSurface, PxrUsdTranslators_PhongReader);
+
+// clang-format off
+TF_DEFINE_PRIVATE_TOKENS(
+    _tokens,
+
+    // Maya material nodes attribute names
+    (cosinePower)
+    (specularColor)
+);
+// clang-format on
+
+PxrUsdTranslators_PhongReader::PxrUsdTranslators_PhongReader(const UsdMayaPrimReaderArgs& readArgs)
+    : PxrUsdTranslators_LambertReader(readArgs)
+{
+}
+
+/* static */
+UsdMayaShaderReader::ContextSupport
+PxrUsdTranslators_PhongReader::CanImport(const UsdMayaJobImportArgs& importArgs)
+{
+    return importArgs.shadingConversion == UsdMayaShadingConversionTokens->phong
+        ? ContextSupport::Supported
+        : ContextSupport::Unsupported;
+}
+
+const TfToken& PxrUsdTranslators_PhongReader::_GetMayaNodeTypeName() const
+{
+    return UsdMayaShadingConversionTokens->phong;
+}
+
+void PxrUsdTranslators_PhongReader::_ConvertToMaya(const TfToken& mayaAttrName, VtValue& usdValue)
+    const
+{
+    if (mayaAttrName == _tokens->cosinePower && usdValue.IsHolding<float>()) {
```

Comment:
Need to convert back roughness to cosinePower using the reverse of the formula used in the exporter.